pkg/operator/controllers/machine_controller.go

```diff
@@ -0,0 +1,252 @@
+package controllers
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	azureproviderv1beta1 "github.com/openshift/cluster-api-provider-azure/pkg/apis/azureprovider/v1beta1"
+	machinev1beta1 "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	clusterapi "github.com/openshift/cluster-api/pkg/client/clientset_generated/clientset"
+	"github.com/operator-framework/operator-sdk/pkg/status"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	aro "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	aroclient "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned/typed/aro.openshift.io/v1alpha1"
+	_ "github.com/Azure/ARO-RP/pkg/util/scheme"
+)
+
+const (
+	machineSetsNamespace = "openshift-machine-api"
+)
+
+// MachineReconciler reconciles the alertmanager webhook
+type MachineReconciler struct {
+	clustercli      clusterapi.Interface
+	arocli          aroclient.AroV1alpha1Interface
+	log             *logrus.Entry
+	developmentMode bool
+	role            string
+}
+
+func NewMachineReconciler(log *logrus.Entry, clustercli clusterapi.Interface, arocli aroclient.AroV1alpha1Interface, role string) *MachineReconciler {
+	return &MachineReconciler{
+		clustercli:      clustercli,
+		arocli:          arocli,
+		log:             log,
+		role:            role,
+		developmentMode: true, //TODO we need this!!
+	}
+}
+
+func vmSizeIsValid(vmSize api.VMSize, developmentMode bool, isMaster bool) bool {
+	if isMaster {
+		switch vmSize {
+		case api.VMSizeStandardD8sV3,
+			api.VMSizeStandardD16sV3,
+			api.VMSizeStandardD32sV3:
+		default:
+			return false
+		}
+	} else {
+		if developmentMode {
```

Comment:
@mjudeikis because of the above ^
I could move the developmentMode into a container ENV to be less public